ledger/test-common/src/main/daml/model/Test.daml

```diff
@@ -682,3 +682,37 @@ template MultiPartyContract
       do
         actualCid <- lookupByKey @MultiPartyContract keyToFetch
         assertMsg "LookupOtherByKey value matches" (expectedCid == actualCid)
+
+template WithKey
+  with
+    p : Party
+  where
+    signatory p
+    key p : Party
+    maintainer key
+    controller p can
+      WithKey_Archive : ()
+        do
+          return ()
+
+template WithKeyCreator
+  with
+    p1 : Party
+    p2 : Party
+  where
+    signatory p1
+    observer p2
+    choice WithKeyCreator_DiscloseCreate : ContractId WithKey
+      with
+        p : Party
+      controller p
+      do create (WithKey p1)
+
+template WithKeyFetcher with
+    fetcher : Party
+    contractId : ContractId WithKey
+  where
+    signatory fetcher
+    controller fetcher can
+      nonconsuming WithKeyFetcher_Fetch: WithKey
+        do fetch contractId
```

Comment:
This will be used in a subsequent PR.

Reply:
Is that to test that the same thing works for divulgence instead of disclosure? Because if not, we should definitely have a testcase for this even if it happens to not be broken at the moment.

Reply:
```
  p1 <- allocatePartyOn "p1" (ParticipantName "A")
  p2 <- allocatePartyOn "p2" (ParticipantName "B")
  helper <- submit p1 $ createCmd (Helper p1 p2)
  cid <- submit p1 $ exerciseCmd helper (DiscloseCreate p1)
  -- fetch f1 by p2 should succeed
  submit p1 $ archiveCmd cid
  -- fetch f2 by p2 should fail
  ```

  That's the scenario that's going to be implemented.